Fix [Can't resolve '@babylonjs/core/Behaviors/Meshes/baseSixDofDragBe…

flyskywhy opened the pull request:
…havior.js'](https://github.com/brianzinn/react-babylonjs/issues/220)

And "@babylonjs/core": "^5.1.0" here is ref to https://github.com/brianzinn/react-babylonjs/blob/v3.1.2/packages/react-babylonjs/package.json#L49

flyskywhy committed:
Fix [Can't resolve '@babylonjs/core/Behaviors/Meshes/baseSixDofDragBehavior.js'](https://github.com/brianzinn/react-babylonjs/issues/220)

flyskywhy committed:
Update to react-babylonjs 3.1.2 and its related BabylonJS exactly 5.1.0

flyskywhy committed:
react -> react-native: just simply merge code from `react-native init GCanvasRNExamples --version 0.63.2`

flyskywhy committed:
react -> react-native: no need `--no-package-lock` any more when `npm install`

brianzinn commented:
There's been a breaking change upstream.  I am investigating - thank you for reporting.

flyskywhy committed:
react -> react-native: fix ESLint error `jsxBracketSameLine is deprecated` with not latest "prettier", and

the reason of not using latest "eslint" is to: fix ESLint `TypeError: Module.createRequire is not a function` when using nodejs v10.15.3

flyskywhy committed:
react -> react-native: works on Android with just an Welcome <View/> wrapped by react-navigation which also need upgrade @babel/core to
fix `SyntaxError: GCanvasRNExamples\node_modules\@react-navigation\core\src\types.tsx: Unexpected token, expected ","`

flyskywhy committed:
react -> react-native: downgrade @react-native-community/eslint-config from 3.0.2 to 3.0.1 to
fix `SublimeLinter: ERROR: TypeError: Error while loading rule 'prettier/prettier': context.getPhysicalFilename is not a function`

flyskywhy committed:
react -> react-native: works on Web with react-native-web; and

remove `react-native/Libraries/NewAppScreen` in `App.js` to
fix [Field 'browser' doesn't contain a valid alias configuration](https://github.com/necolas/react-native-web/issues/2095);

use cross-env to let most package.json scripts can run on Windows;

since DISABLE_ESLINT_PLUGIN=true in `npm run web` , then we can use any version of eslint, so use eslint@7 even can be used in nodejs 10

upgrade eslint from 6.5.1 to 7.8.1 to
fix ESLint `Error: Failed to load plugin 'flowtype' declared in '.eslintrc.js » @react-native-community/eslint-config#overrides[0]': Cannot find module 'eslint/use-at-your-own-risk'`

not upgrade eslint to ^8.4.1 to
fix ESLint `TypeError: Module.createRequire is not a function` when using nodejs v10.15.3

flyskywhy committed:
react -> react-native: let PRODUCT_BUNDLE_IDENTIFIER on iOS same with package name on Android

flyskywhy committed:
react -> react-native: just reorder package.json to react-native like

flyskywhy committed:
react -> react-native: since there is `.eslintrc.js`, so eslintConfig in package.json is useless

flyskywhy committed:
react -> react-native: simple navigate from HomeScreen on Web and Android ref to https://reactnavigation.org/docs/navigating

flyskywhy committed:
react -> react-native: babylonjs `Drag and drop` component works on Web

flyskywhy committed:
react -> react-native: add react-native-unimodules@0.12.0 so comes the dependencies: expo-asset expo-constants expo-file-system expo-image-loader expo-permissions

flyskywhy committed:
react -> react-native: add @flyskywhy/react-native-gcanvas@2.3.25

flyskywhy committed:
react -> react-native: add `Canvas 2d Demo` component

flyskywhy committed:
typo

flyskywhy committed:
react -> react-native: npm install webgl-utils.js m4.js to practice code in [WebGL Fundamentals](https://webglfundamentals.org/webgl/lessons/webgl-fundamentals.html) later

flyskywhy committed:
react -> react-native: `Webgl 3d Textures` step1 just add a blank component

flyskywhy committed:
react -> react-native: `Webgl 3d Textures` step2 just copy code from https://webglfundamentals.org/webgl/lessons/zh_cn/webgl-3d-textures.html

flyskywhy committed:
react -> react-native: `Webgl 3d Textures` step4 works well on Android with new version @flyskywhy/react-native-gcanvas match webglUtils.resizeCanvasToDisplaySize()

flyskywhy committed:
react -> react-native: `Webgl Cube Maps` step1 just add a blank component

flyskywhy committed:
react -> react-native: fix sometimes download err with gradle on Android by adding aliyun to maven

flyskywhy committed:
react -> react-native: fix `D8: Cannot fit requested classes in a single dex file`

flyskywhy committed:
fix React-Native Android import from node_modules directory does not working ref to https://github.com/facebook/react-native/issues/35204#issuecomment-1304740228 caused by
https://github.com/facebook/react-native/issues/35210

flyskywhy committed:
react -> react-native: `Webgl 3d Textures` step3 works well on Web

flyskywhy committed:
react -> react-native: `Webgl Cube Maps` step2 just copy code from https://webglfundamentals.org/webgl/lessons/zh_cn/webgl-cube-maps.html

flyskywhy committed:
react -> react-native: `Webgl Cube Maps` step3 works well on Web

flyskywhy committed:
react -> react-native: `Webgl Cube Maps` step4 works well on Android with new version @flyskywhy/react-native-gcanvas support HTMLCanvasElement to texImage2D

flyskywhy committed:
react -> react-native: `babylonjs Non-Declarative` works well on Android with new version @flyskywhy/react-native-gcanvas support 3d game engine babylonjs

flyskywhy committed:
README.md for GCanvas React Native Examples

flyskywhy committed:
Canvas 2d Demo: fix can not display image after 2nd click "draw some" button

flyskywhy committed:
react -> react-native: add `npm run build-android` script

flyskywhy committed:
Suppress warning 'React Components must start with an uppercase letter'

flyskywhy committed:
react -> react-native: fix `npm run android` on Linux `error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup. Run CLI with --verbose flag for more details. Error: spawn ./gradlew EACCES`

flyskywhy committed:
react -> react-native: fix nodejs 8.9.3 will let `npm run web` cause D:\proj\GCanvasRNExamples\node_modules\terser-webpack-plugin\node_modules\jest-worker\build\WorkerPool.js:25
  } catch {
          ^

SyntaxError: Unexpected token {
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:599:28)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
    at Function.Module._load (module.js:489:3)
    at Module.require (module.js:579:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (D:\proj\GCanvasRNExamples\node_modules\terser-webpack-plugin\node_modules\jest-worker\build\index.js:38:42)

flyskywhy committed:
react -> react-native: support `npm run web-clean` on Windows and avoid error `node_modules\sane\src\node_watcher.js:291:16`
